packages/nimble-components/src/table/models/expansion-manager.ts

```diff
@@ -31,6 +39,9 @@ export class ExpansionManager<TData extends TableRecord> {
             return false;
         }
 
+        if (row.subRows.length === 0) {
+            return false;
+        }
```

Comment:
So this makes it so that after the user expands a row we don't expand immediately (i.e. how it doesn't expand in storybook)?

I think I find the behavior surprising. If you look at the row-expand-toggle event it shows that the row is expanded each time you click it:
![image](https://github.com/ni/nimble/assets/1588923/88f11198-1cbb-4836-ba23-a91ba1c2f29d)

From an event lifecycle that seems surprising, maybe unexpected for an app, how does it go from false -> true without going true -> false in between? And that doesn't match the visual. The newState isn't actually expanded true.

Lets say there is a slow event reaction time (like Blazor server has to respond to the event over the network so the transition to loading is delayed). I think for a user it would be surprising, it might feel broken, they may click it again firing off more events.

I wonder if instead it should be disabled? The representation of a row that is:
- expanded in the `canLoadChildren` state is a disabled expand button
- collapsed in the `canLoadChildren` state is an enabled expand button

Thoughts? If there is more to consider / it's not a straightforward tweak we could avoid blocking the PR and track it in an issue.

Reply:
The current behavior comes from the line in the HLD that says _a row with no children and a `delayedHierarchyState` of `canLoadChildren` will always be collapsed_. The goal with that statement was to handle the cases that (1) moving from `loadingChildren` -> `canLoadChildren` ensures the row's state will be collapsed, and (2) a row with a state of `canLoadChildren` that had children & then the children are removed goes back to being collapsed.

I agree, though, that having the button not change is a bit inconsistent. I'm not sure that disabling the button is the right way to go. Perhaps I can handle the specific cases I mentioned more explicitly rather than always having the row be collapsed. It almost certainly won't be a straightforward tweak to the existing code, so I'll probably address that in a follow up PR.